src/Test/Plutip/Contract.hs

```diff
@@ -295,13 +295,15 @@ withContractAs ::
   TestRunner w e a
 withContractAs walletIdx toContract = do
   (cEnv, wallets') <- ask
-  let wallets@(ownWallet :| otherWallets) = reorder walletIdx wallets'
-  let contract = wrapContract wallets (toContract (map ledgerPaymentPkh otherWallets))
+  let (ownWallet, otherWallets) = separateWallets walletIdx wallets'
+      collectValuesPkhs = fmap ledgerPaymentPkh wallets'
+      otherPkhs = map ledgerPaymentPkh otherWallets
+      contract = wrapContract collectValuesPkhs (toContract otherPkhs)
   liftIO $ runContract cEnv ownWallet contract
   where
-    reorder i xss = case NonEmpty.splitAt i xss of
-      (xs, y : ys) -> y :| xs ++ ys
-      _ -> error $ "Should fail: bad wallet index for own wallet: " <> show i
+    separateWallets i xss
+      | (xs, y : ys) <- NonEmpty.splitAt i xss = (y, xs ++ ys)
```

Comment:
Very minor stylistic thing for consistency since `<>` is used elsewhere

```suggestion
      | (xs, y : ys) <- NonEmpty.splitAt i xss = (y, xs <> ys)
```